### PR TITLE
Add js bundling

### PR DESCRIPTION
**Pull request by PerezIgnacio**
#### Board:
* [Add JS Bundling system to the rails_api_base](https://www.notion.so/Add-JS-Bundling-system-to-the-rails_api_base-a1ea921bae5945438d2b10cb3e63a62c)
---
#### Description:

* This PR adds jsbundling-rails gem, using esbuild as the bundler. Also includes custom config file for esbuild, changes to make it compatible with active_admin/arctic_admin, and some modifications for use with docker.
---
#### Notes:
---

### CODE REVIEW

**Review by Rodrigo-Miranda-0 — APPROVED** [review]
Awesome!

**Review by sebastiancaraballo — APPROVED** [review]

File: Procfile.dev [hunk 1]
```diff
@@ -0,0 +1,2 @@
+web: unset PORT && env RUBY_DEBUG_OPEN=true bin/rails server
```
Comment (santib) [inline Procfile.dev:1]: why `unset PORT` and why `RUBY_DEBUG_OPEN=true` ? do we want to merge this?

File: README.md [hunk 2]
```diff
@@ -80,6 +83,7 @@ To illustrate, `bin/rails console` will run the console in the docker container
 - [Faker](https://github.com/stympy/faker) for generating test data
 - [Flipper](https://github.com/jnunemaker/flipper) for feature flag support
 - [Jbuilder](https://github.com/rails/jbuilder) for json views
+- [JavaScript Bundling for Rails](https://github.com/rails/jsbundling-rails) for bundling JS assets.
```
Comment (santib) [inline README.md:86]:
```suggestion
- [JS Bundling](https://github.com/rails/jsbundling-rails) for bundling JS assets.
```

minor comment, feel free to ignore

File: esbuild.config.mjs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import * as esbuild from 'esbuild'
+
+const config = {
+  entryPoints: ['app/assets/javascripts/*'],
+  bundle: true,
+  sourcemap: process.env.RAILS_ENV != 'production',
+  minify: process.env.RAILS_ENV == 'production',
```
Comment (santib) [inline esbuild.config.mjs:7]:
```suggestion
  sourcemap: process.env.RAILS_ENV !== 'production',
  minify: process.env.RAILS_ENV === 'production',
```

File: package.json [hunk 4]
```diff
@@ -0,0 +1,16 @@
+{
+  "name": "app",
```
Comment (santib) [inline package.json:2]:
```suggestion
  "name": "rails_api_base",
```

so it can be replaced by the script?

File: README.md [hunk 5]
```diff
@@ -34,7 +34,8 @@ This template comes with:
 1. Install PostgreSQL in case you don't have it
```
Comment (santib) [inline README.md:34]: maybe somewhere in the README we need to mention ESBuild, that's used for the admin site, and that?

File: Procfile.dev (hunk 1 above)
Comment (santib) [reply Procfile.dev:1]: what about
`web: bin/rails server -p3000` ?

**Review by santib — APPROVED** [review]
👏

File: Dockerfile [hunk 6]
```diff
@@ -1,11 +1,17 @@
...
 RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
-RUN apt update && apt install yarn
+
+RUN mkdir -p /etc/apt/keyrings
+RUN curl -fsSL https://deb.nodesource.com/gpgkey/nodesource-repo.gpg.key | gpg --dearmor -o /etc/apt/keyrings/nodesource.gpg
+
+ENV NODE_MAJOR=18
+RUN echo "deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_$NODE_MAJOR.x nodistro main" | tee /etc/apt/sources.list.d/nodesource.list
+
+RUN apt update && apt install -y yarn nodejs
```
Comment (JulianPasquale) [inline Dockerfile:14]: We can also install node using binaries https://github.com/nodejs/help/wiki/Installation#how-to-install-nodejs-via-binary-archive-on-linux

File: app/assets/javascripts/jquery.js [hunk 7]
```diff
@@ -0,0 +1,4 @@
+import jquery from 'jquery';
```
Comment (JulianPasquale) [inline app/assets/javascripts/jquery.js:1]: Why is this needed? Is this required by ActiveAdmin?
If the goal here is to be able to use jquery for our custom JS scripts, I'd like to not do that. I think we should start using Stimulus for those scenarios.

File: Procfile.dev (hunk 1 above)
Comment (PerezIgnacio) [reply Procfile.dev:1]: Yes, I think that's better for the port, the other config is not necessary.

File: README.md (hunk 5 above)
Comment (PerezIgnacio) [reply README.md:34]: sure

File: app/assets/javascripts/jquery.js (hunk 7 above)
Comment (PerezIgnacio) [reply app/assets/javascripts/jquery.js:1]: It's used by activeadmin, same for jquery-ui. Some things don't work without it (like batch actions, datepicker, etc.)

**Review by JulianPasquale — APPROVED** [review]

**Review by juanmferreira93 — APPROVED** [review]

File: README.md [hunk 8]
```diff
@@ -32,9 +32,11 @@ This template comes with:
 
 1. Clone this repo
 1. Install PostgreSQL in case you don't have it
+1. Install yarn and node.
```
Comment (santib) [inline README.md:35]:
```suggestion
1. Install node and yarn.
```
